packages/cf-component-table/src/Table.js

```diff
@@ -1,41 +1,90 @@
 import React from 'react';
-
 import PropTypes from 'prop-types';
+import {
+  mapChildren,
+  filterNone,
+  applyTheme,
+  createComponent,
+  connectStyles
+} from 'cf-style-container';
+import TableTheme from './TableTheme.js';
+
+const TableStyles = ({ theme, bordered }) =>
+  filterNone({
+    background: theme.colorWhite,
+    borderCollapse: bordered ? 'separate' : 'collapse',
+    borderSpacing: 0,
+    maxWidth: '100%',
+    width: '100%',
+
+    // Table styles
+    border: bordered ? '1px solid ${theme.tableBorderColor}' : undefined,
+    borderLeft: bordered ? 0 : undefined,
+
+    [`@media (max-width: ${theme.breakpoints.tablet})`]: {
+      display: 'block',
+      width: '100%',
+      boxSizing: 'border-box'
+    },
+
+    [`@media screen and (max-width: ${theme.breakpoints.tablet})`]: {
+      display: 'block'
+    }
+  });
 
 class Table extends React.Component {
   render() {
-    let className = 'cf-table';
-
-    if (this.props.striped) className += ' cf-table--striped';
-    if (this.props.hover) className += ' cf-table--hover';
-    if (this.props.bordered) className += ' cf-table--bordered';
-    if (this.props.condensed) className += ' cf-table--condensed';
-    if (this.props.className.trim())
-      className += ' ' + this.props.className.trim();
-
-    return (
-      <table className={className}>
-        {this.props.children}
-      </table>
+    const {
+      is,
+      className,
+      striped,
+      hover,
+      bordered,
+      condensed,
+      bare,
+      children
+    } = this.props;
+
+    return React.createElement(
+      is,
+      { className },
+      mapChildren(children, child => {
+        if (React.isValidElement(child)) {
+          return React.cloneElement(child, {
+            condensed,
+            striped,
+            bordered,
+            hover,
+            bare
+          });
+        }
+        return child;
+      })
     );
   }
 }
 
 Table.propTypes = {
-  className: PropTypes.string,
+  is: PropTypes.oneOf(['table', 'div']),
+  className: PropTypes.string.isRequired,
   striped: PropTypes.bool,
   hover: PropTypes.bool,
   bordered: PropTypes.bool,
   condensed: PropTypes.bool,
+  bare: PropTypes.bool,
   children: PropTypes.node
 };
 
 Table.defaultProps = {
-  className: '',
+  is: 'table',
   striped: false,
   hover: false,
   bordered: true,
-  condensed: false
+  condensed: false,
+  bare: false
 };
 
-export default Table;
+export const createTable = (...styles) =>
```

Comment:
This should not be exported because it can be used as

```js
import { createTable } from 'cf-component-table';

const Table = createTable(() => { absolutely_any_styles_I_want_to_use_I_can_put_here });
```

So that's exactly same as allowing appending a custom className in CSS world.

On the other hand, TableTheme is directly imported by the component and not exposed to for overwriting at all.

So more restricted and controlled access to customization (theme) is replaced by "do whatever you want to do" `styles` param. **It should be the other way around.**

My believe is that developers should not have that level of access because then the "owners" of cf-ui are completely losing control over the styles and it would be impossible for them to refactor/improve without introducing breakages.